streaming/util/json.py

```diff
@@ -0,0 +1,9 @@
+# Copyright 2022-2024 MosaicML Streaming authors
```

Comment:
Should we name this file as `typing.py`?

Reply:
I figure if we add more typing annotations stuff, like for args shorthand unions, it would live together with the other args shorthand code, so similarly json typing stuff ought to live in a json-specific source file